src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -926,25 +969,29 @@ protected virtual string GetOutputFileName(Compilation compilation, Cancellation
         }
         private Func<string, FileMode, FileAccess, FileShare, Stream> _fileOpen;
 
-        private Stream OpenFile(string filePath,
-                                TextWriter consoleOutput,
-                                FileMode mode = FileMode.Open,
-                                FileAccess access = FileAccess.ReadWrite,
-                                FileShare share = FileShare.None)
+        private void ReportExceptionAsDiagnostic(Exception e, string filePath, TextWriter consoleOutput)
```

Comment:
Can we take it one step further and make this a static method that can be called directly from NoThrowStreamDisposer?